src/Integration/UserSettingsConfiguration/GlobalSettingsStorage.cs

```diff
@@ -0,0 +1,121 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using System.IO;
+using System.IO.Abstractions;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.FileMonitor;
+using SonarLint.VisualStudio.Core.Initialization;
+using SonarLint.VisualStudio.Core.SystemAbstractions;
+using SonarLint.VisualStudio.Core.UserRuleSettings;
+using SonarLint.VisualStudio.Infrastructure.VS.Initialization;
+
+namespace SonarLint.VisualStudio.Integration.UserSettingsConfiguration;
+
+[Export(typeof(IGlobalSettingsStorage))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+internal sealed class GlobalSettingsStorage : IGlobalSettingsStorage
+{
+    private const string SettingsFileName = "settings.json";
+    private const string GeneratedGlobalSettingsFolderName = ".global";
+    private readonly IEnvironmentVariableProvider environmentVariableProvider;
+    private readonly ISingleFileMonitorFactory fileMonitorFactory;
+    private readonly IFileSystem fileSystem;
+    private readonly IAnalysisSettingsSerializer serializer;
+
+    private bool disposed;
+    private (string settingsFile, string generatedConfigsBaseDirectory) globalFilePaths;
+    private ISingleFileMonitor globalSettingsFileMonitor;
+
+    [ImportingConstructor]
+    public GlobalSettingsStorage(
+        ISingleFileMonitorFactory singleFileMonitorFactory,
+        IFileSystemService fileSystem,
+        IEnvironmentVariableProvider environmentVariableProvider,
+        IAnalysisSettingsSerializer serializer,
+        IInitializationProcessorFactory processorFactory)
+    {
+        this.fileSystem = fileSystem;
+        this.environmentVariableProvider = environmentVariableProvider;
+        this.serializer = serializer;
+        fileMonitorFactory = singleFileMonitorFactory;
+        InitializationProcessor = processorFactory.CreateAndStart<GlobalSettingsStorage>(
+            [],
+            () =>
+            {
+                if (disposed)
+                {
+                    return;
+                }
+
+                CreateGlobalSettingsMonitorAndSubscribe();
+            });
+    }
+
+    public IInitializationProcessor InitializationProcessor { get; }
+    public event EventHandler SettingsFileChanged;
+    public string SettingsFilePath => globalFilePaths.settingsFile;
+    public string ConfigurationBaseDirectory => globalFilePaths.generatedConfigsBaseDirectory;
+
+    public void EnsureSettingsFileExists()
+    {
+        if (!fileSystem.File.Exists(SettingsFilePath))
+        {
+            serializer.SafeSave(SettingsFilePath, new GlobalAnalysisSettings());
+        }
+    }
+
+    public void SaveSettingsFile(GlobalAnalysisSettings settings)
+    {
+        serializer.SafeSave(SettingsFilePath, settings);
+        InvokeSettingsFileChanged();
+    }
+
+    public GlobalAnalysisSettings LoadSettingsFile() => serializer.SafeLoad<GlobalAnalysisSettings>(SettingsFilePath);
```

Comment:
If this value is cached here, AnalysisSettings.GlobalFileExclusions can be removed and GlobalSettingsProvider can use this value directly instead of going through UserSettingsProvider